packages/mui-material/src/styles/overrides.d.ts

```diff
@@ -111,14 +112,23 @@ import { TooltipClassKey } from '../Tooltip';
 import { TouchRippleClassKey } from '../ButtonBase/TouchRipple';
 import { TypographyClassKey } from '../Typography';
 
-export type OverridesStyleRules<ClassKey extends string = string> = Record<
+export type OverridesStyleRules<
+  ClassKey extends string = string,
+  ComponentName = keyof ComponentsPropsList,
+  Theme = unknown,
+> = Record<
   ClassKey,
-  CSSInterpolation
+  | CSSInterpolation
+  | ((
+      props: (ComponentName extends keyof ComponentsPropsList
+        ? { ownerState: ComponentsPropsList[ComponentName] }
```

Comment:
This may be tricky as some components have more things inside the `ownerState`. I think we may need to export type for it in each component. We can do it in a follow up, as it should only widen the types. By default we can export the props from each component and change only where necessary